src/views/templates/details/tabs/details/components/CPUMemory.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { useVirtualMachineTemplatesCPUMemory } from 'src/views/templates/list/hooks/useVirtualMachineTemplatesCPUMemory';
+import { isCPUMemoryEditable } from 'src/views/templates/utils';
+
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { TemplateModel, V1Template } from '@kubevirt-utils/models';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template';
+import { useVMIAndPodsForVM } from '@kubevirt-utils/resources/vm';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+
+import CPUMemoryModal from './CPUMemoryModal';
+
+type CPUMemoryProps = {
+  template: V1Template;
+};
+
+const CPUMemory: React.FC<CPUMemoryProps> = ({ template }) => {
+  const { t } = useKubevirtTranslation();
+  const CPUMemData = useVirtualMachineTemplatesCPUMemory(template);
+  const asVM = getTemplateVirtualMachineObject(template);
+  const { vmi } = useVMIAndPodsForVM(asVM?.metadata?.name, asVM?.metadata?.namespace);
```

Comment:
a vm-template is by definition a template that should have a vm object
but the vm object is a template for a vm it's not a vm you can actually run.
in line 29 you are looking for the vmi object a mv is managing, since this is not an actual vm, looking for a vmi is useless.

> I need more specific explanation about what the problem is here. Should I rename the variables or remove some of the logic? Maybe I am missing something.

I don't know what the fix is, I noticed that you are trying to get a vmi out of a template, that does not make sense, so something must be wrong here.
posible fix:
a. call the vm object `vmObject` instead of `asVM`, because `asVM` hint that you have a `union` an object that you can look at as VM or as a Template, but actually the VM is one object in the templates object list.
b. remove the vmi const, check where it's used and why, if it's used it's probably a problem too, so fix this places too.

Reply:
Yes, I agree with using the name _vmObject_ here. Anyway nothing inappropriate happens if `vmi` won't be specified, but I'll better make sure. Wanted to have this code more universal. Thanks for the details, will update the PR accordingly.